### ISSUE TITLE
Define time period of support for IDE tools to support older versions of flutter

### ISSUE THREAD

**Issue by helin24** [flutter/flutter#103216]
This topic came up because we'd like to retire some complex code in the IntelliJ plugin that makes the plugin compatible to very old versions of flutter tools. We'd like to have a specified period of support that we promise users, and then encourage people to update to newer versions.

Some questions mentioned by @nuritzi :
- Timeline: We don't have a particular due date in mind. We need some guidance on this before we start retiring old code.
- What's needed: A defined length of time of support, or some other metric like usage percentage at which we stop supporting old versions of flutter. For example, currently about 0.8% of flutter tools usages are from pre-2.0 versions of flutter, but we still have checks in our plugin code to support these versions.
- Main stakeholders: The IntelliJ plugin and the VSCode plugin would be affected. Internal users would not be affected.

cc @itsjustkevin @timsneath @DanTup @jwren @jacob314

**Comment by nuritzi** [comment]
Thanks, @helin24! 

I found a deprecation template and created e a version of it that we can collaborate on for this project: [Old Flutter IDE Deprecation Plan](https://docs.google.com/document/d/1evCNgj25czZwAjL6jD3fuO9JYOcUK5e6xP-D_Ox6-4E/edit?resourcekey=0-aueVq3BwBC6m9BSjLrhosg#). 

Is there a better name for it? 

I'd love your help adding anything you can to it, especially helping us understand things like how many users would be impacted and any risks or obstacles you see around the migration. 

Maybe you and the team can begin filling that brief out? Then, when I'm back during the week of May 23rd, I can take a look and jump back in!

**Comment by helin24** [comment]
I modified the name a little to clarify that we want to deprecate support for old Flutter SDKs in IDEs (hopefully the name is clear? This is difficult to title!).

I filled in most of the sections and will forward to the other people working on IDEs this week.

**Comment by nuritzi** [comment]
@itsjustkevin adding you to this issue.

**Comment by itsjustkevin** [comment]
Next steps:
- Come up with communication plan.
- Review the viability of dropping versions prior to 2.12
- Utilize the breaking change process.

**Comment by matanlurey** [comment]
Closing as stale. Please feel free to re-open, or open a new issue, if work is still planned.

(To be clear, the `-release` and `-infra` teams have no plans to work on this in the known future)
